packages/core/src/device/ble/PeripheralRequest.ts

```diff
@@ -0,0 +1,99 @@
+// We disable no-await-in-loop here.
+// Because no-await-in-loop intend to let user have full advantage of the parallelization benefits of async/await.
+// But we would like to send them sequentially rather than send them parallel.
+
+/* eslint-disable no-await-in-loop */
+import isNil from 'lodash/isNil';
+import isEmpty from 'lodash/isEmpty';
+import { delay } from '../../utils';
+import Transport from '../../transport';
+import { MCU_FINISH_CODE, COMMAND_FINISH_CODE, PACKET_DATA_SIZE } from '../constants';
+import { byteArrayToHex, hexToByteArray } from './utils';
+
+/**
+ * PeripheralRequest is responsible for peripheral data communication.
+ * @class
+ *
+ * @param {Transport} transport internal transport, which must be specified with different platforms
+ * @param {boolean} isFinish since our device can only accept one kind of command once, we should have a flag to determine whether another command is executing.
```

Comment:
```suggestion
 * @param {boolean} isFinish since the device can only accept one kind of command once, a flag is needed to determine whether another command is executing.
```